### ISSUE TITLE
Axolotl (Signal) app is not working

### ISSUE THREAD

**Issue by Stevie00911** [HelloVolla/ubuntu-touch-beta-tests#159]
Since I have the Volla 22, I suffer from not having a working Signal app.

Actually the only app available in the OpenStore is Axolotl from Aaron Kimmig, last updated 27. Oct 2022.
This app woked hardly on the Vollaphone with many bugs. 

With the Volla 22 the app is not no more workable.
It is not possible:
- to add, edit and import contacts
- to receive messages in groups
- to beginn a new conversation
- to see the name or number of a conversation
And sometimes messages of different senders arrive in one thread.

After a deletion of the registration, an unistall and reinstall, Axolotl does not send a neu registration code to the phone number. 
So the app is acually not insallable.

Is it possible to interact with the programmer to fix the Volla 22 issues, or are there other Signal projects for the Volla 22?

**Comment by gbdomubpkm** [comment]
At this point, the Volla xenial is more mature, so supports everything better, is less buggy than the Volla22 xenial. But as focal is not far away (it remains very vague), Axolotl is not yet focal compatible to my knowledge and from there it works a little (volla, volla22 or other focal compatible smartphones),  it is one problem among many others i think...
Only the developers know but I don't think much time is invested in resolving Axolotl bugs under xenial when focal arrives (in the distance...).

**Comment by JamiKettunen** [comment]
Based on the titles of the issues I see on upstream repo of [Axolotl (Signal)](https://open-store.io/app/textsecure.nanuc) (https://github.com/nanu-c/axolotl/issues) and devices they affect these don't seem to be device specific (Volla Phone 22 exclusive) problems; simply the app is at fault and you will likely encounter the same issues if you reinstall the app on an original Volla Phone (or any other device running Ubuntu Touch) too.

I've contacted the developer but I assume they're already aware of the current problems with the app.

**Comment by JamiKettunen** [comment]
I got a response, https://github.com/nanu-c/axolotl/pull/985 is supposedly going to fix the issues with the app, but it's broken currently due to Signal API changes.

As a temporary workaround I'd recommend using the official Signal Android app inside WayDroid (can be setup through [Waydroid Helper](https://open-store.io/app/waydroidhelper.aaronhafer)) until the above mentioned PR is merged and a new version of the app is available on the OpenStore; it's not ideal due to no notifications showing up in Ubuntu Touch and also a higher battery draw if left running in the background, but at least it works :)

**Comment by Stevie00911** [comment]
Thank you all for your infos.
They give a little hope to look for the next Axolotl update.
@Jami, thank you for the workaround with waydroid. 
I could install it, but did not find a way to install the Signal Android app.
There ist no store, or an app install untility on the waydroid desktop.
Additionally waydroid chrashes often the Volla 22.

**Comment by JamiKettunen** [comment]
Waydroid isn't exactly stable yet so the crashes are unfortunately kind of expected.

Waydroid Helper lets you install with Google apps to get for example the Play Store, it's the small "Google+" icon in top right when installing, it's rather well hidden and probably should be made more obvious

**Comment by Stevie00911** [comment]
Thank you, Jami, for your hints.
Yes, the small "Google+" icon is really small and hidden.
The upcoming warning message, says, that with the Google apps the critical Google environment come to the device.
I avoided to continue installing the Google apps, because the reason, why I have the Volla 22 phone and why I am here in the forum is, to have NOT any Google or Apple environment on my phone. ;-)

**Comment by JamiKettunen** [comment]
In that case downloading the APK directly from https://signal.org/android/apk/ (blue download button) with the browser inside Waydroid should also work. You may also install [F-Droid](https://f-droid.org) as I believe that is installed by default on Volla OS as well.

**Comment by Stevie00911** [comment]
"I got a response, https://github.com/nanu-c/axolotl/pull/985 is supposedly going to fix the issues with the app, but it's broken currently due to Signal API changes."
Hi Jami, any news to this old issue? 
Since Jan 2022 we have no update to this singnal only app for Ubuntu Touch.

**Comment by JamiKettunen** [comment]
The pull request is still being worked on with latest update from 2 weeks ago but you can already use it by building it from source while Volla Phone plugged in with developer mode enabled (and https://clickable-ut.dev/en/latest/install.html done on computer side):
```
git clone https://github.com/nanu-c/axolotl -b axolotl_presage
cd axolotl
clickable build --all --arch arm64
clickable install
```
I've built it just now if you don't want to go through that as all the artifacts from https://github.com/nanu-c/axolotl/actions?query=branch%3Aaxolotl_presage seem to have expired already: 
[textsecure.nanuc_2.0.1_arm64.click.zip](https://github.com/HelloVolla/ubuntu-touch-beta-tests/files/13321799/textsecure.nanuc_2.0.1_arm64.click.zip) (9MB)

I don't use Signal but feel free to report back here if the click I built works as expected! At least it launches on my end

**Comment by Stevie00911** [comment]
Hi Jami, thank you for your update info about the project. Your workaround seems a bit too complicated for me at the moment as a normal Windows user. So I willl wait and donot give up hoping, that the app will be available once in the OpenStore.

**Comment by JamiKettunen** [comment]
If you don't mind using the prebuilt click I attached you simply unzip it to internal storage of the phone, navigate to it in file manager and upon opening it with OpenStore accept installation prompt

**Comment by Stevie00911** [comment]
Hi Jami, thank you very much for the zip file. 
With it I was able to install "Axolotl2" on the Volla Phone 22.
With this beta version the registraton of the phone number was successful.
But the software is still as buggy, as the old version of Axolotl.
It is still not possible:
 - to add, edit and import contacts
 - to receive messages single or in groups
 - to beginn a new conversation
You can do nothing with this app.
Greetings, Stefan

**Comment by JamiKettunen** [comment]
https://open-store.io/app/axolotl.nanuc has been available on 20.04 for ~2 months now and should work as well as the tethered (secondary) Signal clients can, if there's further issues reports probably should go to https://github.com/nanu-c/axolotl/issues since I doubt the developers look in this issue tracker :)
